File: 链表/206.反转链表.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
package 链表
type ListNode struct {
   Val int
   Next *ListNode
}
func reverseList(head *ListNode) *ListNode {
	pre := &ListNode{Val: -1, Next: head}
	cur := head
	pre.Next = nil
	for cur != nil {
		tmp := cur.Next
		cur.Next = pre.Next
		pre.Next = cur
		cur = tmp
	}
	return pre.Next
}

File: 链表/206.反转链表.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode pre = new ListNode(-1,head);
        ListNode cur = head;
        pre.next=null;
        while(cur!=null){
            ListNode tmp = cur.next;
            cur.next=pre.next;
            pre.next=cur;
            cur=tmp;
        }
        return pre.next;
    }
}